From d2b3f630dd8846c7833126763cf3ec7983133906 Mon Sep 17 00:00:00 2001
From: hanwckf <my375229675@gmail.com>
Date: Sat, 27 Jul 2019 21:49:36 +0800
Subject: [PATCH 06/19] aw2013: set default state

---
 drivers/leds/leds-aw2013.c      | 26 +++++++++++++++++++
 1 files changed, 26 insertions(+)

diff --git a/drivers/leds/leds-aw2013.c b/drivers/leds/leds-aw2013.c
index 81a3a7d4b..50b4c4ecd 100755
--- a/drivers/leds/leds-aw2013.c
+++ b/drivers/leds/leds-aw2013.c
@@ -627,6 +627,30 @@ free_err:
 	return rc;
 }

+static int aw2013_led_default_set(struct aw2013_led *led_array,
+				struct device_node *node)
+{
+	struct aw2013_led *led;
+	struct device_node *temp;
+	int parsed_leds = 0;
+	const char* state;
+
+	for_each_child_of_node(node, temp) {
+		led = &led_array[parsed_leds];
+		state = of_get_property(temp, "default-state", NULL);
+		if (state) {
+			if (!strcmp(state, "on")) {
+				aw2013_set_brightness(&led->cdev, led->cdev.max_brightness);
+			} else {
+				aw2013_set_brightness(&led->cdev, LED_OFF);
+			}
+		}
+		parsed_leds++;
+	}
+
+	return 0;
+}
+
 static int aw2013_led_probe(struct i2c_client *client,
 			   const struct i2c_device_id *id)
 {
@@ -681,6 +705,8 @@ static int aw2013_led_probe(struct i2c_client *client,
 		goto fail_parsed_node;
 	}

+	aw2013_led_default_set(led_array, node);
+
 	return 0;

 fail_parsed_node:
--
2.25.1

