From b2733e0b31b45e8cf4f65349da88060f785f17f5 Mon Sep 17 00:00:00 2001
From: hanwckf <my375229675@gmail.com>
Date: Sun, 28 Jul 2019 13:16:05 +0800
Subject: [PATCH 07/19] aw2013: add .shutdown

---
 drivers/leds/leds-aw2013.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/leds/leds-aw2013.c b/drivers/leds/leds-aw2013.c
index 50b4c4ecd..69a5254e1 100755
--- a/drivers/leds/leds-aw2013.c
+++ b/drivers/leds/leds-aw2013.c
@@ -637,6 +637,7 @@ static int aw2013_led_default_set(struct aw2013_led *led_array,

 	for_each_child_of_node(node, temp) {
 		led = &led_array[parsed_leds];
+		led->client = led_array->client;
 		state = of_get_property(temp, "default-state", NULL);
 		if (state) {
 			if (!strcmp(state, "on")) {
@@ -737,6 +738,12 @@ static int aw2013_led_remove(struct i2c_client *client)
 	return 0;
 }

+static void aw2013_led_shutdown(struct i2c_client *client)
+{
+	struct aw2013_led *led_array = i2c_get_clientdata(client);
+	aw2013_write(led_array, AW_REG_LED_ENABLE, 0x00);
+}
+
 static const struct i2c_device_id aw2013_led_id[] = {
 	{"aw2013_led", 0},
 	{},
@@ -752,6 +759,7 @@ static struct of_device_id aw2013_match_table[] = {
 static struct i2c_driver aw2013_led_driver = {
 	.probe = aw2013_led_probe,
 	.remove = aw2013_led_remove,
+	.shutdown = aw2013_led_shutdown,
 	.driver = {
 		.name = "aw2013_led",
 		.owner = THIS_MODULE,
--
2.25.1

